src/utils.lisp

```diff
@@ -97,19 +96,31 @@ accumulated, in the order."
   `(let ((it ,condition))
      (when it ,@body)))
 
-(defun integer-to-string (x)
+(defun integer-to-string (x &optional (radix (or *print-base* 10)) plusp)
```

Comment:
`integer-to-string` is loaded pretty early. I will remove the case for `*print-radix*` and supports just the sign and the base.

The rest is responsibility for the printer. It's also easier then because we have better streams support.
.
Also, note that the radix prexis should be printed before the sign, like in `#x-FF`